pkg/cloudprovider/cloudprovider.go

```diff
@@ -57,17 +57,24 @@ func GetCloudConfig(pconfig providerconfigtypes.Config, kubeletVersion string) (
 	return "", errors.New("unknown cloud provider")
 }
 
-func KubeletCloudProviderConfig(cloudProvider providerconfigtypes.CloudProvider) (inTreeCCM bool, external bool, err error) {
+func KubeletCloudProviderConfig(cloudProvider providerconfigtypes.CloudProvider, external bool) (inTreeCCM bool, OutOdTree bool, err error) {
 	switch osmv1alpha1.CloudProvider(cloudProvider) {
-	case osmv1alpha1.CloudProviderAWS, osmv1alpha1.CloudProviderAzure, osmv1alpha1.CloudProviderGoogle,
+	case osmv1alpha1.CloudProviderAWS, osmv1alpha1.CloudProviderAzure,
 		osmv1alpha1.CloudProviderOpenstack, osmv1alpha1.CloudProviderVsphere:
+		if external {
+			return false, true, nil
+		}
+
 		return true, false, nil
 
-	case osmv1alpha1.CloudProviderKubeVirt:
+	case osmv1alpha1.CloudProviderKubeVirt, osmv1alpha1.CloudProviderDigitalocean, osmv1alpha1.CloudProviderHetzner:
 		return false, true, nil
 
-	case osmv1alpha1.CloudProviderAlibaba, osmv1alpha1.CloudProviderAnexia, osmv1alpha1.CloudProviderDigitalocean,
-		osmv1alpha1.CloudProviderHetzner, osmv1alpha1.CloudProviderLinode, osmv1alpha1.CloudProviderEquinixMetal,
+	case osmv1alpha1.CloudProviderGoogle:
+		return true, false, nil
+
+	case osmv1alpha1.CloudProviderAlibaba, osmv1alpha1.CloudProviderAnexia,
+		osmv1alpha1.CloudProviderLinode, osmv1alpha1.CloudProviderEquinixMetal,
 		osmv1alpha1.CloudProviderScaleway, osmv1alpha1.CloudProviderNutanix, osmv1alpha1.CloudProviderVMwareCloudDirector:
```

Comment:
I would replace this with `return false, external, nil` (but leave GCP out of this).